paddle/fluid/operators/sum_op.h

```diff
@@ -97,11 +97,11 @@ void LodTensorArrayCompute(const framework::ExecutionContext &context) {
     auto &in_array = in_vars[i]->Get<framework::LoDTensorArray>();
 
     for (size_t i = 0; i < in_array.size(); ++i) {
-      if (in_array[i].numel() != 0) {
+      if (in_array[i].IsInitialized() && (in_array[i].numel() != 0)) {
```

Comment:
Note that if the returned value of `numel()` is not 0, it doesn't mean that `in_array[i]` have valid data, the valid data may be deleted as garbage in the previous iteration.